chia/full_node/coin_store.py

```diff
@@ -463,21 +469,32 @@ async def batch_coin_states_by_puzzle_hashes(
                     f'WHERE (cr.puzzle_hash in ({"?," * (puzzle_hash_count - 1)}?) '
                     f'OR h.hint in ({"?," * (puzzle_hash_count - 1)}?)) '
                     f"AND (cr.confirmed_index>=? OR cr.spent_index>=?) "
-                    f"{height_filter} "
+                    f"{height_filter} {amount_filter}"
                     f"ORDER BY MAX(cr.confirmed_index, cr.spent_index) ASC "
                     f"LIMIT ?",
-                    puzzle_hashes_db + puzzle_hashes_db + (min_height, min_height, max_items + 1),
+                    (
+                        puzzle_hashes_db
+                        + puzzle_hashes_db
+                        + (min_height, min_height)
+                        + ((min_amount.to_bytes(8, "big"),) if min_amount > 0 else ())
```

Comment:
did you test that `BLOB`s are correctly compared by `>=` in sqlite?
i.e. byte-by-byte

Reply:
The SQLite docs say
> When two BLOB values are compared, the result is determined using `memcmp()`.

https://www.sqlite.org/datatype3.html